### PR TITLE
ci: Add integration tests

### PR DESCRIPTION
**Pull request by jcvenegas**
Add scripts to run integration test in semaphoreci.

`./.ci/setup.sh` : Install clear containers using clearcontainers/tests setup script
`./.ci/run.sh`   : Build and install agent then run integration test

Fixes: #55

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

### CODE REVIEW

**Comment by grahamwhaley** [comment]
Travis looks happy to me...
lgtm

[Approved with PullApprove](https://pullapprove.com/clearcontainers/agent/pull-request/60/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=clearcontainers/agent) [Approved with PullApprove](https://pullapprove.com/clearcontainers/agent/pull-request/60/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=clearcontainers/agent)

File: .ci/setup.sh [hunk 1]
```diff
@@ -20,16 +20,28 @@
...
+project_dir="${GOPATH}/src/${github_project}"
 
-    # Clone Tests repository.
-    go get "$test_repo"
+# Clone Tests repository.
+go get "$test_repo"
+# Fixme: Remove after https://github.com/clearcontainers/runtime/pull/353
+# is merged
+# Change runtime target
+go get -d github.com/clearcontainers/runtime
```
Comment (sboeuf) [inline .ci/setup.sh]: Make github.com/clearcontainers/runtime a variable as you reuse it later in this script.

File: .ci/setup.sh [hunk 2]
```diff
@@ -20,16 +20,28 @@
...
+pushd ${GOPATH}/src/github.com/clearcontainers/runtime
+sed -i 's/cc-agent.target/clear-containers.target/' create.go
+pushd
```
Comment (sboeuf) [inline .ci/setup.sh]: You need `popd` here, not `pushd`.

File: .ci/setup.sh [hunk 3]
```diff
@@ -20,16 +20,28 @@
...
 
-    test_repo_dir="${GOPATH}/src/${test_repo}"
+# Check agent builds
+make
```
Comment (sboeuf) [inline .ci/setup.sh]: Remove this since the agent is built later in this script.

File: .ci/setup.sh [hunk 4]
```diff
@@ -39,4 +51,20 @@ then
         --body-length 72 \
         --subject-length 75 \
         --verbose
+    # Travis doesn't provide a VT-x environment, so nothing more to do
+    # here.
+    exit 0
 fi
+
+# Check agent builds
+cd go get -d ${github_project} || true
```
Comment (sboeuf) [inline .ci/setup.sh]: You should add a comment about this line. All it does is making sure we have the package dependencies this project needs to be built. Indeed, the agent repo is already on the Semaphore system at the commit version we want to test.
Note this won't move the commit version (which is what we want).

File: .ci/setup.sh [hunk 5]
```diff
@@ -39,4 +51,20 @@ then
...
+cd ${project_dir}
+echo "Build ${github_project}"
+make
```
Comment (sboeuf) [inline .ci/setup.sh]: For performance savings, let's use make -j$(nproc)

File: .ci/setup.sh (hunk 4 above)
Comment (sboeuf) [reply .ci/setup.sh]: Oh and replace your command here:
s/cd go get -d ${github_project} || true/go get -d ${github_project} || true/
You've added one `cd` that we don't need.

File: .ci/setup.sh [hunk 6]
```diff
@@ -39,4 +51,20 @@ then
...
         --subject-length 75 \
         --verbose
+    # Travis doesn't provide a VT-x environment, so nothing more to do
+    # here.
+    exit 0
 fi
+
+# Check agent builds
+cd go get -d ${github_project} || true
+cd ${project_dir}
```
Comment (sboeuf) [inline .ci/setup.sh]: Use `pushd` to be consistent with the rest of this script.
And don't forget `popd` at the end of this block.

File: .ci/setup.sh [hunk 7]
```diff
@@ -39,4 +51,20 @@ then
...
+echo "Build ${github_project}"
+make
+
+pushd "${test_repo_dir}"
+sudo -E PATH=$PATH bash .ci/setup.sh
```
Comment (sboeuf) [inline .ci/setup.sh]: Usually we use `sudo -E PATH=$PATH bash -c ".ci/setup.sh"`

File: .ci/run.sh [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+if [ ! -f "clear-${clr_release}-containers.img" ] ;then
+	unxz clear-${clr_release}-containers.img.xz
+fi
+
+#Mount clear-containers image
+mkdir -p ${MOUNT_DIR}
+loop=$(sudo losetup -f --show "clear-${clr_release}-containers.img")
+sudo partprobe ${loop}
+sudo mount "${loop}p1" "${MOUNT_DIR}"
```
Comment (sboeuf) [inline .ci/run.sh]: You can replace those 3 last lines with:
```bash
sudo mount -o loop,offset=$((2048 * 512)) clear-${clr_release}-containers.img ${MOUNT_DIR}
```

File: .ci/run.sh [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+#build agent and install it in image
+make
```
Comment (sboeuf) [inline .ci/run.sh]: No need to build, you already did that in setup.sh.

File: .ci/run.sh [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+export DESTDIR=${MOUNT_DIR}
```
Comment (sboeuf) [inline .ci/run.sh]: I don't think we need to export this one.

File: .ci/run.sh [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+sudo make install DESTDIR=${MOUNT_DIR}
```
Comment (sboeuf) [inline .ci/run.sh]: Replace with:
```bash
sudo -E PATH=$PATH bash -c "make install DESTDIR=${MOUNT_DIR}"
```

File: .ci/run.sh [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+sudo umount "${MOUNT_DIR}"
+sudo losetup -d "${loop}"
```
Comment (sboeuf) [inline .ci/run.sh]: No need for this line if you replaced the mount with the line I gave you.

File: .ci/run.sh [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+
+#Change conatiner image symlink to use last agent
+sudo ln -sf "$(pwd)/clear-${clr_release}-containers.img" /usr/share/clear-containers/clear-containers.img
+
+# Execute the tests under `clearcontainers/tests` repository.
+pushd "${GOPATH}/src/${test_repo}"
+sudo -E PATH=$PATH make check
```
Comment (sboeuf) [inline .ci/run.sh]: Replace with:
```bash
sudo -E PATH=$PATH bash -c "make check"
```

**Review by sboeuf — CHANGES_REQUESTED** [review]
Several comments. I think that we could move most of the code in run.sh to setup.sh. Actually, the "real" thing that we are running is `make check` from tests repo, therefore I think this should be the only thing in this file.

File: .ci/setup.sh (hunk 1 above)
Comment (jcvenegas) [reply .ci/setup.sh]: done

File: .ci/setup.sh (hunk 2 above)
Comment (jcvenegas) [reply .ci/setup.sh]: done

File: .ci/setup.sh (hunk 3 above)
Comment (jcvenegas) [reply .ci/setup.sh]: done

File: .ci/setup.sh (hunk 4 above)
Comment (jcvenegas) [reply .ci/setup.sh]: done, done

File: .ci/setup.sh (hunk 6 above)
Comment (jcvenegas) [reply .ci/setup.sh]: done

File: .ci/setup.sh (hunk 5 above)
Comment (jcvenegas) [reply .ci/setup.sh]: done

File: .ci/setup.sh (hunk 7 above)
Comment (jcvenegas) [reply .ci/setup.sh]: done

File: .ci/run.sh (hunk 8 above)
Comment (jcvenegas) [reply .ci/run.sh]: I dont think is good idea rely on a offset that should be calculated.

File: .ci/run.sh (hunk 9 above)
Comment (jcvenegas) [reply .ci/run.sh]: done

File: .ci/run.sh (hunk 10 above)
Comment (jcvenegas) [reply .ci/run.sh]: right!  export removed!

File: .ci/run.sh (hunk 11 above)
Comment (jcvenegas) [reply .ci/run.sh]: changed :)

File: .ci/run.sh (hunk 13 above)
Comment (jcvenegas) [reply .ci/run.sh]: done

File: .ci/setup.sh (hunk 1 above)
Comment (jcvenegas) [reply .ci/setup.sh]: ups, just created the var, but did not used , fixing

**Review by sboeuf — APPROVED** [review]

**Pull request merged**